<html>

<head>
<title>Readme</title>
</head>

<body>

<h1>Readme</h1>

Here are some details on how we created the biPheMap app.
<br>
<h2>Datasets</h2>

We used two resources: a) the UK Biobank (UKBB) project; and b) the Genotype-Tissue Expression (GTEx) project. The UKBB project is a prospective EHR-linked cohort with deep genetic and rich phenotypic data collected on approximately 500,000 middle-aged individuals (aged between 40 and 69 years old) recruited from across the United Kingdom 1. The GTEx project is a resource database and associated multi-tissue bank aimed at studying the relationship between genetic variation and gene expression in different human tissues <sup>2</sup>.

<h2>Colocalization method</h2>

We integrated multiple association datasets to assess whether two association signals, one from a genome-wide association study (GWAS) on a phenotype, and the other from expression quantitative trait locus (eQTL) analysis in a tissue, overlap in such a matter that they are consistent with a shared causal gene. This approach, referred to as colocalization, was conducted using coloc2 <sup>3</sup>. The coloc2 method is a Bayesian approach which computes the posterior probability that a genetic variant is both associated with the phenotype and the gene expression level in the tissue. We defined a colocalized signal using PPH4 <span>&#8805;</span> 0.80. 

<h2>GWAS and eQTL summary statistics</h2>

coloc2 requires both GWAS summary data and eQTL association summary data. For GWAS data, we used two sets from the UKBB project. The first set of results are GWAS association test statistics publicly available from the Neale lab (Round 1 in 2,419 phenotypes) <sup>4</sup>. We selected variants with minor allele frequency (MAF) > 0.1% and with association p-value < 5x10<sup>-5</sup>. We further used a second set of UKBB GWAS association statistics computed by the SAIGE testing method. In total, 1,403 case-control phenotypes (PheCodes) were available <sup>5</sup>. We selected variants with MAF > 0.1% and with association p-value < 5x10<sup>-3</sup>. For eQTL association signals, we used data from release V7 of the GTEx project. We restricted our study to the list of 48 tissues (from 620 donors) having a sample size of at least 80. eQTLs with MAF > 1% were considered as input for coloc2. Statistically significant cis-eQTLs were selected as detailed on the GTEx project website <sup>6</sup>.

After further quality control (see details in Rocheleau et al, …), we retained coloc2 results for 496 continuous and binary phenotypes with the Neale GWAS data. With SAIGE data, we generated coloc2 results for 915 case-control (PheCode) phenotypes. In total, the biPheMap contains colocalized signals from 9,151 genes and 1,411 phenotypes across 48 tissues.

<h2>Construction of bipartite networks</h2>

We created a bipartite network for each tissue. In brief, a bipartite network, also called a two-mode network, is a network in which nodes of one mode (i.e. type) are only connected to nodes of the other mode. Associations between phenotypes and genes are indicated by links or edges between them. 

<h2>biLouvain clustering algorithm</h2>

For each tissue, it is expected that the bipartite network of coloc2 results will tend to “cluster” in small groups of related phenotypes with their causally associated genes. To uncover clustering within each network, we applied a newly proposed clustering algorithm, called biLouvain, which extends the well-known unipartite Louvain clustering algorithm. This new algorithm efficiently identifies co-clusters, also called communities, of non-overlapping genes and phenotypes by maximizing a bipartite modularity measure <sup>7</sup>.

<h2>How to explore the biPheMap</h2>

The biPheMap app allows you to explore the content of each co-cluster identified by the biLouvain algorithm in each tissue-level set of colocalized genes and phenotypes. For example, you can list all co-clusters for a given tissue; or search for either a specific gene (or phenotype) and biPheMap will indicate in which co-cluster this gene (or phenotype) appears, along with its co-cluster’s “mates”. Note that the co-cluster’s ID number does not mean anything; it only helps to keep track of the co-cluster at the tissue level.

You can also generate a network plot of the selected co-cluster: for all genes and phenotypes belonging to this co-cluster, links (or edges) indicate a colocalized signal between a gene and a phenotype in the corresponding tissue (but don’t expect to see all genes and phenotypes linked in a selected co-cluster). Remember that, since the clustering is performed tissue by tissue, a gene or a phenotype will only appear in one co-cluster for a given tissue (if that gene or phenotype is colocalized in that specific tissue). Depending on the colocalization signals across the 48 tissues, a specific gene or phenotype may appear in one tissue or in many tissues. 

<h2>References:</h2>
1 UK Biobank:  <a href="https://www.ukbiobank.ac.uk/" target="_blank">https://www.ukbiobank.ac.uk/</a> <br>
2 GTEx:  <a href = "https://gtexportal.org/home/" target="_blank">https://gtexportal.org/home/ </a><br>
3 coloc2: <a href = " https://github.com/Stahl-Lab-MSSM" target="_blank"> https://github.com/Stahl-Lab-MSSM </a><br>
4 Neale’s lab GWAS results: <a href = "http://www.nealelab.is/uk-biobank" target="_blank">  http://www.nealelab.is/uk-biobank </a><br>
5 Lee’s lab SAIGE GWAS results:  <a href ="https://www.leelabsg.org/resources" target="_blank">https://www.leelabsg.org/resources </a><br>
6 <a href = "https://storage.googleapis.com/gtex-public-data/Portal_Analysis_Methods_v7_09052017.pdf" target="_blank">https://storage.googleapis.com/gtex-public-data/Portal_Analysis_Methods_v7_09052017.pdf </a><br>
7 biLouvain:  <a href = "https://github.com/paolapesantez/biLouvain" target="_blank">https://github.com/paolapesantez/biLouvain </a><br>



</body>
</html>
